kover-jvm-agent/src/main/java/kotlinx/kover/jvmagent/KoverJvmAgentPremain.java

```diff
@@ -0,0 +1,119 @@
+
+    static {
+        for (int i = 0; i < regexMetacharacters.length(); i++) {
+            char c = regexMetacharacters.charAt(i);
+            regexMetacharactersSet.add(c);
+        }
+    }
+
+    public static void premain(String argsString, Instrumentation instrumentation) throws Exception {
+        KoverAgentSettings settings = readSettingsFromFile(extractArgsFile(argsString));
+        IntellijIntegration.callPremain(settings, instrumentation);
+    }
+
+    private static File extractArgsFile(String args) {
+        if (!args.startsWith(FILE_PREFIX_IN_ARGS)) {
+            throw new IllegalArgumentException("Incorrect arguments format for Kover JVM agent, arguments are expected to start with " + FILE_PREFIX_IN_ARGS);
+        }
+        return new File(args.substring(FILE_PREFIX_IN_ARGS.length()));
+    }
+
+
+    private static KoverAgentSettings readSettingsFromFile(File file) throws IOException {
+        KoverAgentSettings settings = new KoverAgentSettings();
+
+        BufferedReader reader = null;
+        try {
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), "UTF-8"));
+
+            String line = reader.readLine();
+            while (line != null ) {
+                if (line.startsWith(FILE_PATH_ATTRIBUTE)) {
+                    settings.reportFilePath = line.substring(FILE_PATH_ATTRIBUTE.length());
+                } else if (line.startsWith(EXCLUDE_WITH_WILDCARDS_ATTRIBUTE)) {
+                    String wildcards = line.substring(EXCLUDE_WITH_WILDCARDS_ATTRIBUTE.length());
+                    settings.exclusions.add(wildcardsToRegex(wildcards));
+                } else if (line.startsWith(EXCLUDE_WITH_REGEX_ATTRIBUTE)) {
+                    settings.exclusions.add(line.substring(EXCLUDE_WITH_REGEX_ATTRIBUTE.length()));
+                } else if (line.startsWith(INCLUDE_WITH_WILDCARDS_ATTRIBUTE)) {
+                    String wildcards = line.substring(INCLUDE_WITH_WILDCARDS_ATTRIBUTE.length());
+                    settings.exclusions.add(wildcardsToRegex(wildcards));
+                } else if (line.startsWith(INCLUDE_WITH_REGEX_ATTRIBUTE)) {
+                    settings.exclusions.add(line.substring(INCLUDE_WITH_REGEX_ATTRIBUTE.length()));
+                } else if (line.startsWith(APPEND_ATTRIBUTE)) {
+                    String value = line.substring(APPEND_ATTRIBUTE.length());
+                    if (!isBoolean(value)) {
+                        throw new IllegalArgumentException("Incorrect value for argument " + APPEND_ATTRIBUTE + " in Kover JVM agent arguments file, expected true or false");
+                    }
+                    settings.appendToReportFile = Boolean.parseBoolean(value);
+                } else if (line.length() == 0) {
+                    // skip empty line
+                } else {
+                    throw new IllegalArgumentException("Unrecognized line in Kover arguments file: " + line);
```

Comment:
If args file is supposed to be filled by user (and from commit message I suppose it is), it would be nice to mention in the exception what valid values are. E.g. `Line must start with one of prefixes: ${listOf(FILE_PATH_ATTRIBUTE, APPEND_ATTRIBUTE, ...}`